kratos/geometries/geometry.h

```diff
@@ -1943,6 +1943,64 @@ class Geometry
         return mpGeometryData->IntegrationPoints( ThisMethod );
     }
 
+    /* Creates integration points according to its quadrature rule.
+     * @return integration points.
+     */
+    virtual void CreateIntegrationPoints(
+        IntegrationPointsArrayType& rIntegrationPoints) const
```

Comment:
Just for the record: Here a new IntegrationMethodInfo class will be passed as argument. But as the class definition is not ready yet. The addition will be postponed to a separate PR.